lib/spack/spack/compilers/__init__.py

```diff
@@ -44,12 +43,6 @@
 _other_instance_vars = ['modules', 'operating_system']
 _cache_config_file = []
 
-# TODO: customize order in config file
-if platform.system() == 'Darwin':
-    _default_order = ['clang', 'gcc', 'intel']
-else:
-    _default_order = ['gcc', 'intel', 'pgi', 'clang', 'xlc', 'nag']
```

Comment:
Is the compiler called `xlc` or `xl`? I switched it to `xl` because I'm pretty sure that is the actual name.